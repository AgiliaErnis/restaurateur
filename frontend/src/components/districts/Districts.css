.districts {
  padding: 3.5rem 4rem;
  background: #fff;
}

.districts__container {
  display: flex;
  flex-flow: column;
  align-items: center;
  max-width: 1120px;
  width: 90%;
  margin: 0 auto;
}

.districts__wrapper {
  position: relative;
  margin: 50px 0 0px;
  width: 100%;
}

.districts__items {
  display: flex;
  flex-direction: row;
  margin-bottom: 30px;
  width:100%
}

.districts__item {
  position: relative;
  display: flex;
  flex: 1;
  margin: 0 1rem;
  border-radius: 8px;
}

.districts__item__link {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 50px;
  color: #252e48;
  box-shadow: 0 6px 20px rgba(2, 29, 36, 0.801);
  border-radius: 10px;
  text-decoration: none;
  align-items: center;
  text-align: center;
  justify-content: center;
  background-color: rgb(248, 246, 246);
}

.districts__item__link:hover {
  box-shadow: 0 6px 20px rgba(65, 6, 6, 0.801);
}

h4 {
  padding: 5% 20% 5% 5%;
  color: #252e48;
  font-size: 18px;
}

.districts__item .fa-angle-right {
  padding: 5% 5% 5% 0;
}

.districts h1 {
  font-weight: 400;
}

@media screen and (max-width: 1150px) {
    h4 {
      padding: 5% 15% 5% 3%;
    }

    .fa-angle-right {
      padding: 5% 3% 5% 0;
    }
}

@media screen and (max-width: 1024px) {
   .districts__items {
      display: flex;
      flex-direction: column;
      margin-bottom: 0px;
    }

   h4 {
      padding: 5% 30% 5% 5%;
    }

   .fa-angle-right {
      padding: 5% 5% 5% 0;
    }

   .districts__wrapper {
      width: 420px;
    }

    .districts__item {
      margin-bottom: 15px;
    }
}

@media only screen and (max-width: 720px) {
    .districts {
      padding: 0 2.5rem 2.5rem 2.5rem;
    }
}

@media only screen and (max-width: 520px) {
    .districts__wrapper {
      width: 90%;
    }

    h4 {
      padding: 5% 14% 5% 3%;
    }

    .fa-angle-right {
      padding: 5% 3% 5% 0;
    }
}

@media only screen and (max-width: 400px) {
    .districts__wrapper {
      width: 250px;
    }

    h4 {
      padding: 5% 10% 5% 3%;
      font-size: 16px;
    }

    .fa-angle-right {
      padding: 5% 3% 5% 0;
    }
}