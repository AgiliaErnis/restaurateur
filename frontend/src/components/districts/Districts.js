import React from 'react';
import DistrictItem from './District_item';
import './Districts.css';

function Districts() {
  return (
    <div className='districts'>
      <h1>Popular districts in <b>Prague</b></h1>
      <div className="districts__container">
        <div className="districts__wrapper">
          <ul className="districts__items">
            <DistrictItem
              district="Prague 1"
              num_of_restaurants="50"
            />
            <DistrictItem
              district="Prague 2"
              num_of_restaurants="50"
            />
            <DistrictItem
              district="Prague 3"
              num_of_restaurants="50"
            />
        </ul>
        <ul className="districts__items">
            <DistrictItem
              district="Prague 4"
              num_of_restaurants="50"
            />
            <DistrictItem
              district="Prague 5"
              num_of_restaurants="50"
            />
            <DistrictItem
              district="Prague 6"
              num_of_restaurants="50"
            />
          </ul>
          </div>
        </div>
    </div>
  );
}

export default Districts;
