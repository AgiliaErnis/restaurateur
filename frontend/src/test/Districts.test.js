import * as React from "react";
import * as ReactDOM from "react-dom";
import Districts from "../components/districts/Districts";
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('testing CollegeSection', () => {

    it('should render', () => {
        const DistrictsComponent = renderer.create(<Districts />).toJSON();
        expect(DistrictsComponent).toMatchSnapshot();
    })

})