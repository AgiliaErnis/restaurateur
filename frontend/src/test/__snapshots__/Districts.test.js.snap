// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollegeSection should render 1`] = `
<div
  className="districts"
>
  <h1>
    Popular districts in 
    <b>
      Prague
    </b>
  </h1>
  <div
    className="districts__container"
  >
    <div
      className="districts__wrapper"
    >
      <ul
        className="districts__items"
      >
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 1
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 2
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 3
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
      </ul>
      <ul
        className="districts__items"
      >
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 4
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 5
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
        <li
          className="districts__item"
        >
          <a
            className="districts__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <h4>
              Prague 6
               (
              50
               places)
            </h4>
            <i
              class="fas fa-angle-right"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
